### ISSUE TITLE
setting a field that contains `{smth}` first to a scalar, then to `{}`, will "restore" the `{smth}`

### ISSUE THREAD

**Issue by eallik** [amark/gun#514]
so basically once a field has been set to an object, until you refresh the browser, at any time you set the field to a scalar and then to an empty object, gun will point the field to the original object.

<img width="485" alt="screen shot 2018-03-12 at 12 12 39" src="https://user-images.githubusercontent.com/17533/37278052-2cbd9446-25ef-11e8-8303-11a68714f89f.png">

after refresh:

<img width="473" alt="screen shot 2018-03-12 at 12 16 35" src="https://user-images.githubusercontent.com/17533/37278085-40f58338-25ef-11e8-9151-a8005eb65436.png">

**Comment by amark** [comment]
@eallik yes, this looks wrong. Thank you for reporting it, I very much appreciate it. It will go on the list of things that need to be fixed.

Also, deep apologies again for it taking so long for me to get to these GitHub issues... the 1.5 months ago was insanely stressful me, and after those business deals closed, I had a ton of other important things to deal with, and still catching up now.
